File: abc/abc023/b.js
'use strict';
process.stdin.setEncoding("utf8");

var lines = []; 
var reader = require("readline").createInterface({
  input: process.stdin,
  output: process.stdout,
});

reader.on("line", (line) => {
  lines.push(line);
});
reader.on("close", () => {
    //ここに処理を記述
    let n = parseInt(lines[0], 10);
    let s = lines[1];
    let a = "b";
    if (n === 1) {
      console.log(0);
      process.exit();
    }
    for (let i = 1; i <= n; i++) {
      if (i%3 === 1) {
        a = "a" + a + "c";
      } else if (i%3 === 2) {
        a = "c" + a + "a";
      } else {
        a = "b" + a + "b";
      }
      if (a === s) {
        console.log(i);
        process.exit();
      }
    }
    console.log(-1)
});

File: abc/abc023/b.py
n = int(input())
s = input()
now = "b"
if now == s:
    print(0)
else:
    for i in range(n):
        if i%3 == 0:
            now = "a" + now + "c"
        elif i%3 == 1:
            now = "c" + now + "a"
        else:
            now = "b" + now + "b"
        if now == s:
            print(i+1)
            exit()
    print(-1)